environment

```diff
@@ -30,6 +30,14 @@ _select () {
 # https://github.com/boto/boto3/issues/1179#issuecomment-569940530
 #
 _preauth () {
+	gcloud config set project "$GOOGLE_PROJECT"
+	if ! gcloud auth application-default print-access-token >/dev/null 2>&1; then
+		gcloud auth application-default login
+		if ! gcloud auth application-default print-access-token >/dev/null 2>&1; then
+			>&2 echo "Login failed"
+			exit 1
```

Comment:
Since this is a function, and the script is sourced, `exit` terminates the shell. It would have been easy to test this by temporarily inverting the condition.